### ISSUE TITLE
Can you add README?

### ISSUE THREAD

**Issue by qpmnh** [inspiros/mvda#1]
I use discriminative analisys network for my recent works. But I find a problem that the function torch.eig() is not autograd in Pytorch. Can you share some details about your code ?

Thanks. Hope to get your reply!

**Comment by qpmnh** [comment]
Have you published related papers about this code? If yes, can you share your paper with me?

**Comment by inspiros** [comment]
Hello, sorry for late response. I do not check github frequently, just push the code here from terminal.
This repo is my playground, it is just reimplementation of discriminant analysis algorithms from several papers where they don't share code or their code are a mess of matlab.
I don't have a paper yet, I am baking new algorithms here, got some ideas that worked, hope I can come up with an interesting paper.
I don't really understand your question yet. Definitely, `torch.eig` is not automatic differentiable, only `torch.symeig` does if your input is symmetrical.
As I understand, conventional discriminant analysis algorithms rely on **Generalized Eigen Problem** so it requires you to use `torch.eig` to compute sorted eigen vectors, which is the argmax of the optimization problem. But if you integrate it into a neural network that can be optimized with gradient descent, you don't need to call `torch.eig` at all. What I do is to replace the regular losses (`torch.nn.CrossEntropy`, etc.) with the new Fisher loss:
```python
# Code to compute Sw and Sb scatter matrices
loss = torch.trace(Sw) / torch.trace(Sb)
loss.backward()
optim.step()
```

**Comment by inspiros** [comment]
I might be wrong, just jumped into this field for several months. I don't know the idea of your _discriminative analysis network_, but these 3 functions are all extremely computationally expensive that should never be put into the training process of a neural network:
- Matrix inversion
- Eigen decomposition
- Singular value decomposition
Share me the papers about that _discriminative analysis network_ please, I am keen to learn :D
